equinox/nn/__init__.py

```diff
@@ -34,4 +34,8 @@
 )
 from ._rnn import GRUCell as GRUCell, LSTMCell as LSTMCell
 from ._spectral_norm import SpectralNorm as SpectralNorm
-from ._stateful import State as State, StateIndex as StateIndex
+from ._stateful import (
+    State as State,
+    StatefulLayer,  # noqa: F401
```

Comment:
A bit unsure about this.

Reply:
You need `StatefulLayer as StatefulLayer`. The `X as X` pattern is used to include that it's public API, despite being imported into this file.